### PR TITLE
Migrating Donut chart to v9

### PR DESCRIPTION
**Pull request by srmukher**
image

### CODE REVIEW

File: packages/react-components/react-charts-preview/src/utilities/colors.ts [hunk 1]
```diff
@@ -130,14 +130,14 @@ const getThemeSpecificColor = (colors: string[], isDarkTheme: boolean): string =
 
 export const getNextColor = (index: number, offset: number = 0, isDarkTheme: boolean = false): string => {
   const colors = QUALITATIVE_COLORS[(index + offset) % QUALITATIVE_COLORS.length];
-  return getThemeSpecificColor(colors, isDarkTheme);
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/utilities/colors.ts:133]: revert this change

File: packages/react-components/react-charts-preview/package.json [hunk 2]
```diff
@@ -69,7 +69,8 @@
     "@types/react": ">=16.14.0 <19.0.0",
     "@types/react-dom": ">=16.9.0 <19.0.0",
     "react": ">=16.14.0 <19.0.0",
-    "react-dom": ">=16.14.0 <19.0.0"
+    "react-dom": ">=16.14.0 <19.0.0",
+    "@fluentui/react": "^8.118.0"
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/package.json:73]: remove fluentui/react dependency. Why is it needed

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.styles.ts [hunk 3]
```diff
@@ -1,29 +1,50 @@
...
-      fill: 'transparent',
-    },
-    arcLabel: {
-      fontSize: tokens.fontSizeBase100,
-      fontWeight: tokens.fontWeightSemibold,
-      fill: theme.palette.neutralPrimary,
-    },
+  },
+  focusRing: {
+    strokeWidth: '4px',
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.styles.ts:29]: add back stroke and fill styles

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.styles.ts [hunk 4]
```diff
@@ -1,29 +1,50 @@
...
+    fill: 'transparent',
+  },
+  arcLabel: {
+    fontSize: tokens.fontSizeBase100,
+    fontWeight: tokens.fontWeightSemibold,
+  },
+});
+
+/**
+ * Apply styling to the Carousel slots based on the state
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.styles.ts:39]: fix typo

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.tsx [hunk 5]
```diff
@@ -1,134 +1,128 @@
...
+
+    React.useEffect(() => {
+      _updateChart(props);
+    }, [props]);
+
+    function _onFocus(data: IChartDataPoint, id: string): void {
+      props.onFocusCallback!(data, id, currentRef.current);
+    }
+
+    function _hoverOn(data: IChartDataPoint, mouseEvent: React.MouseEvent<SVGPathElement>): void {
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.tsx:29]: call this function _onHover

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.types.ts [hunk 6]
```diff
@@ -1,10 +1,6 @@
-import { IStyle, ITheme } from '@fluentui/react/lib/Styling';
+import { ITheme } from '@fluentui/react/lib/Styling';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.types.ts:1]: remove ITheme import

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.tsx [hunk 7]
```diff
@@ -1,16 +1,348 @@
...
+    const [isPopoverOpen, setPopoverOpen] = React.useState(false);
+
+    React.useEffect(() => {
+      if (_rootElem) {
+        setWidth(_rootElem.offsetWidth);
+        setHeight(_rootElem.offsetHeight - LEGEND_CONTAINER_HEIGHT);
+      }
+    }, []);
+
+    // React.useEffect(() => {
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.tsx:59]: add todo about what is the issue here

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.types.ts [hunk 8]
```diff
@@ -1,8 +1,8 @@
-import { IStyle } from '@fluentui/react/lib/Styling';
-import { IRenderFunction, IStyleFunctionOrObject } from '@fluentui/react/lib/Utilities';
+import { IRenderFunction } from '@fluentui/react/lib/Utilities';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.types.ts:1]: remove IRenderFunction and StringAxis imports.

File: packages/react-components/react-charts-preview/src/components/DonutChart/Pie/Pie.tsx [hunk 9]
```diff
@@ -1,110 +1,101 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable react/jsx-no-bind */
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import * as React from 'react';
-import { pie as d3Pie } from 'd3-shape';
-import { IPieProps, IPieStyleProps, IPieStyles } from './index';
+import { PieArcDatum, pie as d3Pie } from 'd3-shape';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/Pie/Pie.tsx:5]: where is PieArcDatum used?

File: packages/react-components/react-charts-preview/src/components/Legends/Legends.tsx [hunk 10]
```diff
@@ -1,14 +1,419 @@
...
-  LegendsBase,
-  getStyles,
+
+import { IContextualMenuItem } from '@fluentui/react/lib/ContextualMenu';
+import { HoverCard, HoverCardType, IExpandingCardProps } from '@fluentui/react/lib/HoverCard';
+import { classNamesFunction, find, getNativeProps, buttonProperties } from '@fluentui/react/lib/Utilities';
+import { ResizeGroup } from '@fluentui/react/lib/ResizeGroup';
+import { IProcessedStyleSet } from '@fluentui/react/lib/Styling';
+import { OverflowSet, IOverflowSetItemProps } from '@fluentui/react/lib/OverflowSet';
+import { Button } from '@fluentui/react-components';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/Legends/Legends.tsx:9]: import Button from from '@fluentui/react-button' directly.

File: packages/react-components/react-charts-preview/src/components/Legends/Legends.types.ts [hunk 11]
```diff
@@ -1,5 +1,4 @@
 import * as React from 'react';
-import { ITheme, IStyle } from '@fluentui/react/lib/Styling';
 import { IStyleFunctionOrObject } from '@fluentui/react/lib/Utilities';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/Legends/Legends.types.ts:2]: remove import for IStyleFunctionOrObject

Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/Legends/Legends.types.ts:2]: keeping for now, as hover card styles are dependent on this and its breaking on removing this import. Will remove it when the hover card is migrated.

File: packages/react-components/react-charts-preview/src/utilities/colors.ts (hunk 1 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/utilities/colors.ts:133]: done

File: packages/react-components/react-charts-preview/package.json (hunk 2 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/package.json:73]: removed

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.styles.ts (hunk 3 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.styles.ts:29]: done

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.tsx (hunk 7 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.tsx:59]: done

File: packages/react-components/react-charts-preview/src/components/DonutChart/Pie/Pie.tsx (hunk 9 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/DonutChart/Pie/Pie.tsx:5]: it is used during debugging, removed

File: packages/react-components/react-charts-preview/src/components/Legends/Legends.tsx [hunk 12]
```diff
@@ -1,14 +1,419 @@
...
+      let defaultSelectedLegends = {};
+      if (props.canSelectMultipleLegends) {
+        defaultSelectedLegends =
+          props.defaultSelectedLegends?.reduce((combinedDict, key) => ({ [key]: true, ...combinedDict }), {}) || {};
+      } else if (props.defaultSelectedLegend) {
+        defaultSelectedLegends = { [props.defaultSelectedLegend]: true };
+      }
+
+      setSelectedLegends(defaultSelectedLegends);
+    }, [props]);
```
[the remark this answers is not in the document]
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/Legends/Legends.tsx:67]: done

File: packages/react-components/react-charts-preview/src/components/Legends/Legends.tsx (hunk 10 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/Legends/Legends.tsx:9]: done

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.types.ts (hunk 6 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.types.ts]: done

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.types.ts (hunk 8 above)
Comment (srmukher) [reply packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.types.ts]: done

**Review by AtishayMsft — APPROVED** [review]
